utils/HelpAbstractionGenerator/ReadMe.md

```diff
@@ -0,0 +1,47 @@
+##Creating the webHelpAbstraction.json file
```

Comment:
Thanks for adding this file!

Reply:
Well, it was a reasonable assumption for me (at this point) that it would be asked for... lol

I only finished the C#, I still have to update some stuff to hookup the help system for the DCL filetype.

Still struggling with trying to get the JSON file copied to the right location in the vsix. If you look at these 2 copy functions from the pack.py, I don't understand what this could accomplish. I followed the one that was already there, but mine looks like its targeting itself by doing that. I have tried putting it in ./out/src/help/ however that isn't the path that the TS is trying to read and the build or install process spawned a bunch of (i18 related) like-named jsons with it. This thing doesn't want to naturally copy over and VSCode is doing weird things with it. I don't suppose we could just call it a .txt and revisit dumping it somewhere that isn't going to fight me tooth and nail?

```python
def copyAcadProcFinder():
    src = os.path.join(os.path.curdir, 'utils', 'acadProcessFinder', 'bin', 'acadProcessFinder.exe')
    dst = os.path.join(os.path.curdir, 'out', 'process')
    copyFile(src, dst, 'copied acadProcessFinder.exe')

def copyWebHelpAbstraction():
    src = os.path.join(os.path.curdir, 'extension', 'src', 'help', 'webHelpAbstraction.json')
    dst = os.path.join(os.path.curdir, 'extension', 'src', 'help')
    copyFile(src, dst, 'copied webHelpAbstraction.txt')
```
I also took the src/** .vscodeignore and expanded it, but I think my src/* is probably still blocking everything in the folder. I had hoped that the single * would only block the immediate src files and let me manage the subdirectories.
```
src/*
src/help/openWebHelp.md
src/help/openWebHelp.ts
src/completion/**
src/format/**
src/process/**
src/project/**
src/test/**
```

Reply:
I am not sure if this works:` I had hoped that the single * would only block the immediate src files and let me manage the subdirectories.` If it can work, it is great. Otherwise I suggest to move the webHelpAbstraction.json to` Extension/Data` folder.

Reply:
Little surprised to see that TypeScript didn't have any native support for conditional build statements. So, I dirtied the function signature and built my own version of it. The code below lets us keep the file in /src/help/ and still generates a vsix with a working reference to the data file. If this is acceptable, shall I include my changes to Pack.py in the commit?

FYI, the added (i18) json files were for my ts->js not the copied file.

My *openWebHelp.ts*
```typescript
export function readAllHelpData() {
	loadWebHelpAbstraction(
		"../../extension/src/help/webHelpAbstraction.json",
		"../../out/help/webHelpAbstraction.json");
}
function loadWebHelpAbstraction(srcFile: string, outFile: string): void {
	webHelpContainer = new WebHelpLibrary();
	var fs = require("fs");
	var resolvedPath = path.resolve(__dirname, outFile);
	if (fs.existsSync(resolvedPath) === false) {
		resolvedPath = path.resolve(__dirname, srcFile);	
	}
    fs.readFile(resolvedPath, function(err, data) {        
        if (err === null) {                      
            webHelpContainer.load(JSON.parse(data));
        }
    });    
}
```
The *Pack.py*
```python
def copyWebHelpAbstraction():
    src = os.path.join(os.path.curdir, 'extension', 'src', 'help', 'webHelpAbstraction.json')
    dst = os.path.join(os.path.curdir, 'out', 'help', 'webHelpAbstraction.json')
    copyFile(src, dst, 'copied webHelpAbstraction.json')
```

Reply:
Maybe this is an option: `come to think of it, I could also just add the same /out/help/ copy to the winCompile.bat and only reference that path in the TS. `

And please think about to put the webHelpAbstraction.json in `extension/data`, and other data files are also in that folder. I don't want to have a black sheep.

Reply:
I think the complications with not having it there totally validates what your saying, but this is just where I was told to put it https://github.com/Autodesk-AutoCAD/AutoLispExt/pull/20#pullrequestreview-492208349
> Can you please also move `webHelpAbstraction.json` into the help folder? 

Either way, I'll put it wherever now that I know how to work around it.